server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -712,6 +713,8 @@ protected Node(
                 repositoriesServiceReference::get,
                 rerouteServiceReference::get
             );
+            List<IdentityAwarePlugin> identityAwarePlugins = pluginsService.filterPlugins(IdentityAwarePlugin.class);
```

Comment:
I don't think it is going to work: the `createComponents` much later (see please line 977), so the `IdentityPlugin` will not be initialized (no thread pool etc) and as such - won't be able to provide the plugin subject instance

Reply:
ah, good catch! I have been holding off starting the security side of this PR until the changes were closer to being final. This would have been caught if I prepared a security PR.

Reply:
Moved this initialization to after createComponents.